### PR TITLE
Pre-commit and logging improvements, resolve errors

### PR DESCRIPTION
**Pull request by shenef**
~~Draft cause we need to resolve existing errors before new commits can be accepted with the new config~~

### CODE REVIEW

**Comment by jlward** [comment]
Once this gets merged, we can hook in the flake8 plugin to reject prints. But we aren't there yet.

**Comment by shenef** [comment]
Has been validated up to guards (latest defend fixes should allow runs to go past that), I'll keep doing runs until this gets merged.

File: .pre-commit-config.yaml
Comment (jlward) [inline .pre-commit-config.yaml:28]: We'll eventually want to remove this, but for now, it's fine.

Comment (jlward) [reply .pre-commit-config.yaml:28]: Actually the rest after this

File: area/besaid.py
Comment (jlward) [inline area/besaid.py:45]: Super important change here. Lol

File: image_to_text.py
Comment (jlward) [inline image_to_text.py:38]: It's better to make this an `r` string rather than add the `noqa`: `r"$@B%8&WM#*oahkbdpqwmZO0QLCJUYXzcvunxrjft/\|()1{}[]?-_+~<>i!lI;:,\"^`'."` Maybe also a `"""` string too so we don't need to escape the `"`

File: load_game.py
Comment (jlward) [inline load_game.py:563]: It might make more sense to pull these up. But testing is a thing that isn't really done well here. So feel free to ignore this one.

File: memory/main.py
Comment (jlward) [inline memory/main.py:236]: I recall an hour ago seeing this and deciding not to bring it up. Lol. I'm glad you rolled it in.

File: memory/main.py
Comment (jlward) [inline memory/main.py:399]: Do we need this at all? I could be mistaken, but as I understand we need to use `pbar` in order to make use of this. @orkaboy would know better. They added this to begin with

File: tracker/ffx_rng_tracker/data/equipment_shops.py
Comment (jlward) [inline tracker/ffx_rng_tracker/data/equipment_shops.py:56]: The real fix to this is to use `os.path.join`. Feel free to do hold off until a follow up PR.

File: .pre-commit-config.yaml
Comment (shenef) [reply .pre-commit-config.yaml:28]: Yes, they are mostly inconsequential and mainly there to get the important parts of flake8 into the pre-commit workflow

File: tracker/ffx_rng_tracker/data/equipment_shops.py
Comment (shenef) [reply tracker/ffx_rng_tracker/data/equipment_shops.py:56]: This is actually part of the RNG tracker "library" we are using, so i was a bit hesitant to make that change but flake8 kept complaining about it (invalid escape sequence \e or something like that).

**Comment by shenef** [comment]
Validated with a full any% run.

**Review by jlward — APPROVED** [review]

**Pull request merged**
